prusti-viper/src/encoder/pure_function_encoder.rs

```diff
@@ -1345,9 +1361,17 @@ impl<'p, 'v: 'p, 'tcx: 'v> BackwardMirInterpreter<'tcx>
                                 let array_types = self.encoder.encode_array_types(place_ty).with_span(span)?;
                                 state.substitute_value(&opt_lhs_value_place.unwrap(), array_types.array_len.into());
                             }
+                            ty::TyKind::Slice(..) => {
+                                let snap_len = self.encoder.encode_snapshot_slice_len(
+                                    place_ty,
+                                    self.encode_place(place).with_span(span)?.0,
+                                ).with_span(span)?;
+
+                                state.substitute_value(&opt_lhs_value_place.unwrap(), snap_len);
+                            }
                             _ => {
                                 return Err(SpannedEncodingError::unsupported(
-                                    "checking the length of anything but arrays in pure code is not implemented yet",
+                                    "checking the length of anything but arrays and slices in pure code is not implemented yet",
```

Comment:
I guess based on [`Rvalue::Len` documentation](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/enum.Rvalue.html#variant.Len), the variant should never be emitted for anything other than arrays or slices?

Reply:
hm, true. so go with `SpannedEncodingError::internal` then? or full-on `unreachable!()`?

Reply:
`unreachable!` should be fine, with an appropriate comment.